ibis/expr/analysis.py

```diff
@@ -774,28 +776,15 @@ def predicate(node):
 
 def find_subqueries(node: ops.Node) -> Counter:
     counts = Counter()
-
-    def finder(node: ops.Node):
-        if isinstance(node, ops.Join):
-            return [node.left, node.right], None
-        elif isinstance(node, ops.PhysicalTable):
-            return g.halt, None
-        elif isinstance(node, ops.SelfReference):
-            return g.proceed, None
-        elif isinstance(node, (ops.Selection, ops.Aggregation)):
-            counts[node] += 1
-            return [node.table], None
-        elif isinstance(node, ops.TableNode):
-            counts[node] += 1
-            return g.proceed, None
-        elif isinstance(node, ops.TableColumn):
-            return node.table not in counts, None
-        else:
-            return g.proceed, None
-
-    # keep duplicates so we can determine where an expression is used
-    # more than once
-    list(g.traverse(finder, node, dedup=False))
+    for graph in map(g.Graph.from_dfs, util.promote_list(node)):
+        dependents = graph.invert()
+        for u, vs in dependents.toposort().items():
+            # count the number of table-node dependents on the current node
+            # but only if the current node is a selection or aggregation
+            if isinstance(u, (rels.Projection, rels.Aggregation)):
```

Comment:
So `u` here is the subquery and `vs` are the parent nodes depending on the subquery?

Reply:
Yep